packages/server/modules/core/services/users.js

```diff
@@ -156,20 +155,50 @@ module.exports = {
    * @returns {Promise<import('@/modules/core/helpers/types').UserRecord | null>}
    */
   async getUserById({ userId }) {
-    const user = await Users().where({ id: userId }).select('*').first()
+    const user = await Users()
```

Comment:
It really sucks that this service does so much repository-layer query building and while I get that all of that already was there before your changes, this PR is an opportunity to at least get rid of some of that. At the very least not make it worse, which is what's happening as the new queries you're introducing in the service are even bigger and more  complicated than what was there before.

Not just that but these seem to largely duplicate what's already existing in the repo layer and also what's existing in other functions in this same service.

So why not just make these service calls invoke the repo functions instead?

Reply:
I agree with all this.
My only issue is that we have a deadline to respect, and refactoring this module looks like a long journey.
Also consider that we will need to change this logic anyway during the multi-region refactor.

Reply:
Ok, as a compromise lets leave it for future changes, but its really not great that not only this already shitty service is not improved, but actually made more complicated. Maybe leave a TODO for cleanup